devito/ir/iet/visitors.py

```diff
@@ -166,7 +166,9 @@ def _args_call(self, args):
         ret = []
         for i in args:
             try:
-                if i.is_LocalObject:
+                if isinstance(i, Call):
+                    ret.append(self.visit(i).text)
```

Comment:
.text ?

Reply:
It's a field of the cgen object. Perhaps it would be easier to use `str`

Reply:
ah. But why not just ``self.visit(i)`` then?

Reply:
That does not return a string

Reply:
Hmm, so here's the thing:
if using the `text` field, the semicolon is not appended even if it is a statement. This would make the `is_statement` field useless. Using `str` the cgen element is rendered properly and therefore the `is_statement` becomes necessary. What do you reckon would be better?

Reply:
I **think** you have to use ``is_statement``?  Same reason I thought we should have used ``self.visit(i)`` -- how can this work (ie, with ``.text``) if we have, say, triply nested Calls? Maybe I'm missing something...would triply nested Calls work the current way?

Another possibility is that we move part of the logic inside ``visit_Call``

Reply:
It works just fine using `.text` with however many nested calls (we are using `self.visit(i)` first after all). My only reservation is that it is a cgen internal field

Reply:
oh ignore me...ignore me...I'm stupid

I think Im happy with whatever makes you happier. I don't really have a strong opinion here